### ISSUE TITLE
switching graphviz engine doesn't work when in error state

### ISSUE THREAD

**Issue by RBhupi** [sverweij/atom-graphviz-preview-plus#34]
While using clusters in Dot, fdp engine gives error when one object is in two cluster. 

Error: node "match" is contained in two non-comparable clusters "cluster_1" and "cluster_2"

I can understand that fdp does not want the same object in two clusters. So I want to move on and go back to using dot engine again. But It gets stuck in the fdp. Can't change it at all. I have tried re-opening and renaming the file. One error in dot file and it becomes unusable forever.

**Comment by sverweij** [comment]
You can change the layout engine in two other spots
- in the menu: `Packages -> GraphViz preview+ -> Select layout engine`
- in the graphviz-preview-plus settings:
  - Open atom settings
  - Navigate to `packages`
  - Select the atom-graphviz-preview-plus package
  - The 6th setting from the top is the Layout Engine

:arrow_right:  Could you share the offending dot file - I'd like to reproduce the problem (and it would help me help you in case the above steps don't work for you ...

#### from the menu:
![image](https://user-images.githubusercontent.com/4822597/41582859-1315c2d2-73a3-11e8-8bdd-934f0ae5d867.png)

#### in graphviz preview plus settings:
![image](https://user-images.githubusercontent.com/4822597/41582694-9bc9811e-73a2-11e8-8341-84c4352e4f51.png)

**Comment by RBhupi** [comment]
The second option worked for me.
I am using dot - graphviz version 2.38.0 (20140413.2041) on ubuntu 16.04.
I am attaching the dot file. You may be able to reproduce the bug as follow.

1. open it in atom editor
2. Run preview by pressing Ctr+Shift+G
3. Right click on the graph area and choose layout engine fdp.
4. This will show error mentioned earlier.
5. Now you again Right Click and choose layout engine dot. This wont work.
6 restart the atom editor or rename the file, default layout engine is set
to fdp. It wont change.
7. only way to change it is through the settings as you suggested.

Thanks for the reply.

Bhupendra Raut

**Comment by sverweij** [comment]
I've reproduced it. In general, when the previewer is in an _error_ state, switching engines via the context or global menu does not work.

I found the culprit as well - it should be an easy fix

<details>

```javascript
function setEngine(pEngine) {
  // the !this.svg prevents engine switching in _error_ state => we should probably
  // remove that
  if (this.loading || !this.svg) {
    return;
  }
  atom.config.set('graphviz-preview-plus.layoutEngine', pEngine);
  this.emitter.emit('did-change-title');
}
```

</details>

**Comment by sverweij** [comment]
@RBhupi It's fixed in version 1.5.5 (published a few minutes ago on atom.io), so you don't need to switch via the settings anymore.

Thanks for raising this issue!
